src/Chapter3.hs

```diff
@@ -971,6 +1134,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Day -> Bool 
+isWeekend day
+    | elem day [Friday, Saturday, Sunday] = True 
+    | otherwise = False
+
+nextDay :: Day -> Day
+nextDay day
+    | day == maxBound = minBound
+    | otherwise = succ day
+
+daysToParty :: Day -> Int 
+daysToParty day
+    | fromEnum day > fromEnum Friday = fromEnum Friday - fromEnum day + fromEnum (maxBound :: Day) + 1
+    | otherwise = fromEnum Friday - fromEnum day
```

Comment:
That's a great implementation! 👏🏻 

If you remember how `mod` works for negative numbers, you can implement this function even shorter 🙂